src/Esfa.Vacancy.Register.Api/Orchestrators/VacancyOrchestrator.cs

```diff
@@ -1,24 +1,32 @@
-﻿using Esfa.Vacancy.Register.Application.Queries.GetVacancy;
-using MediatR;
-using System.Threading.Tasks;
+﻿using System.Threading.Tasks;
 using AutoMapper;
+using Esfa.Vacancy.Register.Application.Queries.GetVacancy;
+using Esfa.Vacancy.Register.Infrastructure.Settings;
+using MediatR;
 
 namespace Esfa.Vacancy.Register.Api.Orchestrators
 {
-    public class VacancyOrchestrator
+    public class VacancyOrchestrator : IVacancyOrchestrator
     {
         private readonly IMediator _mediator;
+        private readonly string _liveVacancyBaseUrl;
 
-        public VacancyOrchestrator(IMediator mediator)
+        public VacancyOrchestrator(IMediator mediator, IProvideSettings provideSettings)
         {
             _mediator = mediator;
+            _liveVacancyBaseUrl = provideSettings.GetSetting(ApplicationSettingConstants.LiveVacancyBaseUrl);
         }
 
         public async Task<Vacancy.Api.Types.Vacancy> GetVacancyDetailsAsync(int id)
         {
             var response = await _mediator.Send(new GetVacancyRequest() { Reference = id });
+            var vacancy = response == null ? null : Mapper.Map<Vacancy.Api.Types.Vacancy>(response.Vacancy);
 
-            return response == null ? null : Mapper.Map<Vacancy.Api.Types.Vacancy>(response.Vacancy);
+            if (!string.IsNullOrEmpty(_liveVacancyBaseUrl))
```

Comment:
More of a discussion point.  We need to encapsulate the setting of the vacancyUrl so it happens whenever we get a vacancy not just in the GetVacancyDetails method.  I know Sachin's latest check-in is doing some manipulation in the mapper.AfterMap which will happen whenever we need to get a vacancy but I'm wondering whether we should add a GetMappedVacancy() function to the orchestrator which performs the automapper first and then manipulates/adds further details information to the Vacancy afterwards.